installers/source/setup.sh

```diff
@@ -84,18 +37,163 @@ echo_with_color() {
   esac
 }
 
+## Puts text in the center of the terminal, just for layout / making things pretty
+print_centered() {
+  [[ $# == 0 ]] && return 1
+
+  declare -i TERM_COLS="$(tput cols)"
+  declare -i str_len="${#1}"
+  [[ $str_len -ge $TERM_COLS ]] && {
+      echo "$1";
+      return 0;
+  }
+
+  declare -i filler_len="$(( (TERM_COLS - str_len) / 2 ))"
+  [[ $# -ge 2 ]] && ch="${2:0:1}" || ch=" "
+  filler=""
+  for (( i = 0; i < filler_len; i++ )); do
+      filler="${filler}${ch}"
+  done
+
+  printf "%s%s%s" "$filler" "$1" "$filler"
+  [[ $(( (TERM_COLS - str_len) % 2 )) -ne 0 ]] && printf "%s" "${ch}"
+  printf "\n"
+
+  return 0
+}
+
+## Used for each of the individual components to be installed
+run_process () {
+  while true; do echo -n . >&5; sleep 1; done &
+  trap 'kill $BGPID; exit' INT
+  BGPID=$!
+  echo -n "$1" >&5
+  $2
+  echo done >&5
+  kill $!
+}
+
+clear
+
 # Make sure script run as sudo
 if ((EUID != 0)); then
-  echo -e "${RD}\nPlease run script with root privileges: sudo bash $0 \n${NC}" >&5
+  echo -e "${RD}\nPlease run script with root privileges: sudo ./dfsetup.run \n"
   exit 1
 fi
 
+print_centered "-" "-"
+print_centered "-" "-"
+print_centered "Welcome to DreamFactory!"
+print_centered "-" "-"
+print_centered "-" "-"
+print_centered "Thank you for choosing DreamFactory. Please select any additional installation options from the menu choices below (you may select more than one), or simply press enter to continue with the default installation (Nginx Webserver, latest version of DreamFactory).\n" 
+print_centered "-" "-"
+echo -e ""
+echo -e "[1] Install driver and PHP extensions for Oracle DB" 
+echo -e "[2] Install driver and PHP extensions for IBM DB2" 
+echo -e "[3] Install driver and PHP extensions for Cassandra DB" 
+echo -e "[4] Install Apache2 web server for DreamFactory (Instead of Nginx)" 
+echo -e "[5] Install MariaDB as the default system database for DreamFactory" 
+echo -e "[6] Install a specfic version of DreamFactory" 
+echo -e "[7] Run Installation in debug mode (logs will output to /tmp/dreamfactory_installer.log)\n"
+print_centered "-" "-"
+echo -e "E.g input 1,4 and press Enter to additionaly install Oracle Drivers and use Apache2 instead of Nginx" 
+print_centered "-" "-"
+echo_with_color magenta "Input additional installation options or leave blank for default installation and press Enter to continue"
```

Comment:
What are additional installation options?

Reply:
As above, I think the text needs rewording to make it a bit more user friendly